core/trino-spi/src/main/java/io/trino/spi/eventlistener/ViewLikeInfo.java

```diff
@@ -13,150 +13,10 @@
  */
 package io.trino.spi.eventlistener;
 
-import com.fasterxml.jackson.annotation.JsonCreator;
-import com.fasterxml.jackson.annotation.JsonProperty;
-
-import java.util.Optional;
-
-import static io.trino.spi.eventlistener.ViewType.COLUMN_MASK;
-import static io.trino.spi.eventlistener.ViewType.MATERIALIZED_VIEW;
-import static io.trino.spi.eventlistener.ViewType.ROW_FILTER;
-import static io.trino.spi.eventlistener.ViewType.VIEW;
-
 /**
  * This class is JSON serializable for convenience and serialization compatibility is not guaranteed across versions.
  */
-public class ViewLikeInfo
+public interface ViewLikeInfo
```

Comment:
Additionally we can use `JsonTypeInfo` and annotate all implementation so the for deserialization it could be easy. We can implement something similar to `ValueSet`

Reply:
This looks good to me. Can we squash it with the previous commit ? Can we rename it as `ReferenceInfo` or something similar ?

Reply:
I used `TableReferenceInfo` and renamed all of the subclasses accordingly (e.g. `ViewReferenceInfo`). Did not know about `JsonTypeInfo` / `JsonSubTypes`, I thought I would have to implement a full `JsonTypeResolver`. Super nifty, thank you for the tip :)

Squashed as well.